src/aws-util/src/kinesis.rs

```diff
@@ -40,15 +45,19 @@ pub async fn kinesis_client(
                 region,
             )
         }
-        (_, _) => {
+        (None, None) => {
             info!("AWS access_key_id and secret_access_key not provided, creating a new Kinesis client using a chain provider.");
             let mut provider = ChainProvider::new();
             provider.set_timeout(Duration::from_secs(10));
+            provider.credentials().await?; // ensure that credentials exist
```

Comment:
Is this change meant to be part of the same pr?

Reply:
Ah, sorry I meant to mention that this PR builds on #5194 and only the last two commits are meaningful :-(